hoodie-spark/src/main/scala/com/uber/hoodie/HoodieSparkSQLWriter.scala

```diff
@@ -0,0 +1,256 @@
+/*
+ *  Copyright (c) 2017 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+package com.uber.hoodie
+
+import java.util
+import java.util.concurrent.ConcurrentHashMap
+import java.util.Optional
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+
+import com.uber.hoodie.DataSourceWriteOptions._
+import com.uber.hoodie.common.table.HoodieTableMetaClient
+import com.uber.hoodie.common.util.{FSUtils, TypedProperties}
+import com.uber.hoodie.config.HoodieWriteConfig
+import com.uber.hoodie.exception.HoodieException
+import com.uber.hoodie.hive.{HiveSyncConfig, HiveSyncTool}
+import org.apache.avro.generic.GenericRecord
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.hadoop.hive.conf.HiveConf
+import org.apache.log4j.LogManager
+import org.apache.spark.api.java.JavaSparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, SQLContext, SaveMode}
+
+import scala.collection.mutable
+
+private[hoodie] object HoodieSparkSQLWriter {
```

Comment:
Good point. Some thoughts came up about re-using the schema between microbatches, but I checked the other common streaming sinks, like kafka sink / filestreamsink, none of them re-use the schema.
There may be something else we may share between the microbatches, like HoodieWriteClient, KeyGenerator, etc. Might need to think more about that.

Reply:
Please let me know your thoughts here

Reply:
re-using schema would be also problematic in dealing with new fields added in later batches.. I think the other sinks are doing the right thing and we should follow? 

I think the timeline information is something we can share between batches and keep it fresh on the driver.. @bvaradar and I discussed something for #488 . if we decide to do it, it will take care of this anyway